msteams-platform/m365-apps/specify-runtime-requirements.md

```diff
@@ -0,0 +1,143 @@
+
+```json
+    "elementRelationshipSet": {
+      "mutualDependencies" : [
+        [
+                {"name" : "bots", "id" : "bot-id"}, 
+                {"name" : "staticTabs", "id" : "staticTab-id"},
+                {"name" : "composeExtensions", "id" : "composeExtension-id"},
+                {"name" : "configurableTabs", "id": "configurableTab-id"}
+        ]
+      ]
+    },
+```
+
+## Specify runtime requirements for specific app capabilities (`requirementSet`)
+
+Within individual app capability definitions, you can specify specific TeamsJS runtime requirements using a [`requirementSet`](../resources/schema/manifest-schema-dev-preview.md#statictabsrequirementset). This ensures that the app capability only loads in Microsoft 365 hosts with support for the critical TeamsJS capabilities. 
+
+The following TeamsJS capabilities can be specified as runtime requirements for [`staticTabs`](../resources/schema/manifest-schema-dev-preview.md#statictabsrequirementset), [`composeExtensions`](../resources/schema/manifest-schema-dev-preview.md#composeextensionsrequirementset), and [`bots`](../resources/schema/manifest-schema-dev-preview.md#botsrequirementset):
+
+- HTML-based dialogs ([`dialog.url`](/javascript/api/@microsoft/teams-js/dialog.url))
+- HTML-based dialogs for Bot Framework ([`dialog.url.bot`](/javascript/api/@microsoft/teams-js/dialog.url.bot))
+- Adaptive Card dialogs ([`dialog.adaptiveCard`](/javascript/api/@microsoft/teams-js/dialog.adaptivecard))
+- Adaptive Card dialogs for Bot Framework ([`dialog.adaptiveCard.bot`](/javascript/api/@microsoft/teams-js/dialog.adaptivecard.bot))
+
+The following JSON snippet shows a static tab that requires its host to support HTML dialogs (referred to as task modules in TeamsJS v1.x) invoked from tabs and bots: 
+
+```json
+    "staticTabs": [
+        {
+            "entityId": "idForPage",
+            "name": "Display name of tab",
+            "contentUrl": "https://contoso.com/content?host=msteams",
+            "contentBotId": "Specifies to the app that tab is an Adaptive Card Tab. You can either provide the contentBotId or contentUrl.",
+            "websiteUrl": "https://contoso.com/content",
+            "scopes": [
+                "personal"
+            ],
+            "requirementSet": {
+                "hostMustSupportFunctionalities": [
+                  {"name": "dialogUrl"},
+                  {"name": "dialogUrlBot"}
+                ]
+            }
+        }
+    ],
+```
+
+## Code samples
+
+| Sample name | Description | JavaScript |
+|----------------|-----------------|--------------|
+| ElementRelationshipSet | This sample app demonstrates how to specify runtime requirements, such as one-way or mutual dependencies and app capabilities, to tailor app experience across different Microsoft 365 hosts. | [View](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-hello-world/nodejs) |
```

Comment:
```suggestion
| RT Sample App | This sample app demonstrates how to specify runtime requirements for one-way dependencies and app capabilities, and tailor the app experience across different Microsoft 365 hosts. | [View](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/requirement-targeting-oneway-dependency/nodejs) |
| RT-Sample-App | This sample app demonstrates how to specify runtime requirements for mutual dependencies and app capabilities, and tailor the app experience across different Microsoft 365 hosts. | [View](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/requirement-targeting-mutual-dependency/nodejs) |
```

Reply:
Tweaked your suggestion slightly with further details in the names / descriptions.